grpclog/component_test.go

```diff
@@ -0,0 +1,85 @@
+/*
+ *
+ * Copyright 2020 gRPC authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package grpclog
+
+import (
+	"fmt"
+	"reflect"
+	"testing"
+)
+
+func parseAndCompare(t *testing.T, envVar string, envVars, preVars map[string]*ComponentData) {
+	envVarsResult, preVarsResult := parseEnvironmentVar(envVar)
+	if !reflect.DeepEqual(envVars, envVarsResult) {
```

Comment:
https://pkg.go.dev/github.com/google/go-cmp/cmp?tab=doc#Equal to compare
https://pkg.go.dev/github.com/google/go-cmp/cmp?tab=doc#Diff to show the difference

To compare unexported fields, you will need `cmp.Equal(a, b, cmp.AllowUnexported(ComponentData))`

Reply:
Cool package! Testing uses go-cmp now for equivalence and comparison.